tests/VariantUnboxedValuePred.fir

```diff
@@ -0,0 +1,9 @@
+f[Boxed[t]](x: t):
+    let v = ~x
```

Comment:
The design is still not quite right. It's strange that I can't give type to this because `[t, ..r]` is not allowed.

If I'm allowed to infer `[t, ..r]` then I should be able to type it explicitly as well.